[13563] Release lifecycle (#2459)
eProsima/Fast-DDS: RELEASE_SUPPORT.md

# Release support

Please, refer to the [master branch](https://github.com/eProsima/Fast-DDS/blob/master/RELEASE_SUPPORT.md) for the latest version of this document.

*eProsima Fast DDS* maintains several releases with different support cycles.
Major releases are scarce and they are made when public API is broken (more information about *eProsima Fast DDS* versioning can be found [here](https://github.com/eProsima/Fast-DDS/blob/master/VERSIONING.md)).

> Latest major release is v2.0.0 (June 2020) which introduced DDS compliant API.

Reviewer: ```suggestion
Latest major release is v2.0.0 (June 2020), which introduced the DDS compliant API.
```

Each quarter, a new Fast DDS minor version with new features is released.
By default, *eProsima Fast DDS* minor releases have a lifecycle of **6 months**.
However, the life of certain branches can be extended depending on costumer and user needs.

> Within the support period of any minor version, there may be several patch releases that either add new functionalities in an ABI compatible way, or fix possible issues.

Reviewer: ```suggestion
Within the support period of any minor version, there may be several patch releases that either add new functionalities in an ABI compatible way or fix possible issues.
```

> Before the minor release end of life (EOL) date, a final patch release including the latest fixes will be make available if there are any changes not included in any previous patch release of the same minor version.

Reviewer 2: ```suggestion
Before the minor release end of life (EOL) date, a final patch release including the latest fixes will be made available if there are any changes not included in any previous patch release of the same minor version.
```

# Release support

Please, refer to the [master branch](https://github.com/eProsima/Fast-DDS/blob/master/RELEASE_SUPPORT.md) for the latest
version of this document.

*eProsima Fast DDS* maintains several releases with different support cycles.
Major releases are scarce and they are made when public API is broken.

> Latest major release is v2.0.0 (June 2020) which introduced DDS compliant API.

Reviewer: I'd add a link to the versioning policy document somewhere here

> Each quarter, a new Fast DDS minor version is released, including new features.

Reviewer: ```suggestion
Each quarter, a new Fast DDS minor version with new features is released.
```

> By default, *eProsima Fast DDS* minor releases have a lifecycle of 6 months.

Reviewer: ```suggestion
By default, *eProsima Fast DDS* minor releases have a lifecycle of **6 months**.
```

> However, the life of certain branches can be extended depending on costumer needs.

Reviewer: ```suggestion
However, the life of certain branches can be extended depending on costumer and user needs.
```

> Within the support period of any minor version, there can be patch releases that either add new fuctionalities in an

Reviewer: ```suggestion
Within the support period of any minor version, there may be several patch releases that either add new functionalities in an
```

> ABI compatible way, or fix possible issues.

Reviewer: In general, only break lines after period in md

> Before the minor release end of life (EOL), a final patch release including the latest fixes will be done if there are

Reviewer: ```suggestion
Before the minor release end of life (EOL) date, a final patch release including the latest fixes will be make available if there are any
```

> changes not included previously in any other patch release.

Reviewer: ```suggestion
changes not included any previous patch release of the same minor version.
```

The table below outlines the *eProsima Fast DDS* minor releases and their support cycles:

|Version branch|Release Date|EOL Date|
|--------------|------------|--------|
|v1.7.x|December 2018|February 2022|
|v1.8.x|May 2019|February 2022|
|v1.9.x|August 2019|February 2022|
|v1.10.x|April 2020|February 2022|
|v2.0.x|June 2020|February 2022|
|v2.1.x|November 2020|May 2023|
|v2.2.x|January 2021|February 2022|
|v2.3.x|March 2021|November 2022|
|v2.4.x|September 2021|March 2022|
|v2.5.x|December 2021|June 2022|

> |v2.6.x|March 2022|May 2023 [^*]|

Reviewer: I'd order this from most recent to oldest

> [^*]: This release support could be extended.

Reviewer: ```suggestion
[^*]: Support may be extended.
```
